pyqir-jit/src/jit.rs

```diff
@@ -27,22 +31,30 @@ pub fn run_module_file(
         .to_str()
         .expect("Did not find a valid Unicode path string")
         .to_owned();
-    let context_type = ContextType::File(&path_str);
-    let context = BareContext::new(&ctx, context_type)?;
+
+    let module_type = ModuleType::File(&path_str);
+    let context = Bare::new(&ctx, module_type)?;
     run_module(&context.module, entry_point)
 }
 
-pub fn run_module<'ctx>(
-    module: &Module<'ctx>,
-    entry_point: Option<&str>,
-) -> Result<SemanticModel, String> {
-    Target::initialize_native(&InitializationConfig::default()).unwrap();
+/// # Errors
+///
+/// Will return `Err` if LLVM native target fails to initialize
+pub fn run_module(module: &Module<'_>, entry_point: Option<&str>) -> Result<SemanticModel, String> {
+    Target::initialize_native(&InitializationConfig::default())
+        .expect("Failed to initialize native target.");
+
     let default_triple = TargetMachine::get_default_triple();
     let target = Target::from_triple(&default_triple).expect("Unable to create target machine");
-    assert!(target.has_asm_backend());
-    assert!(target.has_target_machine());
 
-    run_basic_passes_on(&module);
+    if !target.has_asm_backend() {
+        return Err("Target doesn't have an ASM backend.".to_owned());
```

Comment:
It looks like these Err cases may not be included in the `# Errors` section above?